### PR TITLE
fix hack for batch size

### PR DESCRIPTION
**Pull request by kuba-krj**
# Description
After refactors dataloaders return tuples, also loss is returned as tuple. I had to make changes that acount for it to make `hack_for_batch_size` work. I also removed decoding in the step 0 (it takes unnecessarily long time in case of hack for batch size)

# Checklist
Answer each question with "yes", and provide an explanation if the answer isn't what is expected.

1. Is this PR focused on a single feature without the possibility of being divided into smaller PRs?
   - Answer: yes
2. Are all variables set with logical defaults that are backward compatible?
   - Answer: yes
3. Can you provide a link to a Neptune experiment verifying the functionality of this PR?
   - Answer: [Link to Neptune](https://app.neptune.ai/o/pmtest/org/llm-efficiency/runs/details?viewId=98a55dc0-aa23-4c1a-a622-93ec989d9ab3&detailsTab=charts&shortId=LLMEF-5801&type=run&lbViewUnpacked=true&sortBy=%5B%22sys%2Fcreation_time%22%5D&sortFieldType=%5B%22datetime%22%5D&sortFieldAggregationMode=%5B%22auto%22%5D&sortDirection=%5B%22descending%22%5D&suggestionsEnabled=true&query=%28%28%60sys%2Fowner%60%3Astring+%3D+%22kuba-krj%22%29%29)
4. Have you successfully executed the linter and tests?
   - Answer:yes
5. Are all functions concise and don't require splitting?
   - Answer:yes
6. Is there documentation for arguments and complex logic?
   - Answer:yes
7. Are there no temporary solutions in the code? Should a task be added to Trello to refine the code?
   - Answer:no tmp sols
8. Are all functions placed in appropriate files?
   - Answer:yes
9. Is the PR name meaningful and descriptive?
   - Answer:yes
10. Is PR description provided, or unnecessary?
    - Answer:provided
  
Other questions:

11. Have you not made changes to the requirements or anything related to the Singularity image? If you have, did you generate and upload a new image to the clusters?
    - Answer:no changes
12. Is the change significant enough to notify all individuals working with this repository?
    - Answer:no
13. Is there a need for a second reviewer?
    - Answer:no

### CODE REVIEW

File: research/conditional/utils/conditional_trainer.py [hunk 1]
```diff
@@ -354,13 +358,13 @@ def _hack_for_batch_size(
             self.train_dataloader.get_batch()
         )
         for tensor in processed_batch:
-            tensor.data = tensor[:1].repeat(step + 1, 1).data
+            tensor[1].data = tensor[1][:1].repeat(step + 1, 1).data
```
Comment (crewtool) [inline research/conditional/utils/conditional_trainer.py:361]:
```suggestion
        for _cool_name_for_this, selfexplainatoryname in processed_batch:
            selfexplainatoryname.data = selfexplainatoryname[:1].repeat(step + 1, 1).data
```

File: research/conditional/utils/conditional_trainer.py [hunk 2]
```diff
@@ -354,13 +358,13 @@ def _hack_for_batch_size(
...
         loss = self._calculate_loss(
             batch=processed_batch,
             model=self.model,
             mixed_precision=self.mixed_precision,
             vocab_size=self.vocab_size,
-        )
+        )[0]
```
Comment (crewtool) [inline research/conditional/utils/conditional_trainer.py:367]:
```suggestion
        loss, _aux_info = self._calculate_loss(
            batch=processed_batch,
            model=self.model,
            mixed_precision=self.mixed_precision,
            vocab_size=self.vocab_size,
        )
```

**Review by crewtool — CHANGES_REQUESTED** [review]

**Review by maciek-pioro — COMMENTED** [review]
Added a question about a change orthogonal to the main purpose of this PR. Also, I agree with @crewtool 's comments

File: research/conditional/utils/conditional_trainer.py [hunk 3]
```diff
@@ -134,7 +134,11 @@ def train(self, n_steps: int):
             if step % 1000 == 0:
                 print(f"Step {step}")
 
-            if self.model_type == "gpt" and step % self.decoding_logging_steps == 0:
+            if (
+                self.model_type == "gpt"
+                and step % self.decoding_logging_steps == 0
+                and step > 0
+            ):
```
[the remark this answers is not in the document]
Comment (kuba-krj) [reply research/conditional/utils/conditional_trainer.py:141]: No, removing

File: research/conditional/utils/conditional_trainer.py (hunk 2 above)
Comment (kuba-krj) [reply research/conditional/utils/conditional_trainer.py:367]: Right, changed

**Review by Simontwice — APPROVED** [review]
